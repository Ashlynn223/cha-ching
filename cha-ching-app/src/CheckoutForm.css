.header {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 30px;
    font-weight: 700;
    padding-bottom: 40px;
    padding-top: 40px;
    text-align: center;
  }
  
  .paymentFormContainer {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  
  .paymentElementcontainer {
    padding-bottom: 20px;
  }
  
  .button {
    background-color: thistle;
    border-radius: 16px;
    cursor: pointer;
    padding: 4px 16px;
  }
  
  .emailLabel {
    margin-bottom: 3px;
  }
  
  .emailInput {
    border: 1.5px solid grey;
    border-radius: 5px;
    margin-bottom: 15px;
    margin-right: 10px;
    padding: 10px;
    width: 93%;
  }
