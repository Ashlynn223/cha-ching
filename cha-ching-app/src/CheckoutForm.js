import {
  PaymentElement,
  useStripe,
  useElements,
} from "@stripe/react-stripe-js";

function CheckoutForm() {
  const stripe = useStripe();
  const elements = useElements();

  const handleSubmit = async () => {
    await stripe
      .confirmPayment({
        elements,
        confirmParams: {
          return_url: "http://localhost:3000/success",
        },
      })
      .then((data) => console.log("What am I (data):", data))
      .catch((err) => console.log("I am an error", err));
  };

  return (
    <div>
      <div className="header">Cha-ching</div>
      <div className="paymentFormContainer">
        <form name="payment-form">
          <div className="emailLabel">Email</div>
          <input className="emailInput"></input>
          <div>
            <PaymentElement />
          </div>
        </form>
        <button onClick={handleSubmit} className="button">
          Submit
        </button>
      </div>
    </div>
  );
}

export default CheckoutForm;
