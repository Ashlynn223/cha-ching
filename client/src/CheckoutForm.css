.checkoutFormBackground {
  background-image: url(./checkoutFormBackground.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  position: relative;
}

.checkoutFormHeader {
  color: #42434d;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 30px;
  font-weight: 700;
  padding-bottom: 20px;
  padding-top: 40px;
  position: relative;
  text-align: center;
}

.price {
  color: rgba(32, 30, 30, 0.6);
  font-family: Arial, Helvetica, sans-serif;
  font-size: 20px;
  font-weight: 600;
  text-align: center;
}

.paymentFormContainer {
  display: flex;
  justify-content: center;
  padding: 20px;
}

.paymentForm {
  align-items: center;
  background-color: rgba(255, 255, 255, 0.6);
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 20px;
  width: 40vw;
}

.paymentElementcontainer {
  padding-bottom: 20px;
  width: 100%;
}

.nameAndEmailInputsContainer {
  width: 100%;
}

.nameAndEmailInputs {
  display: flex;
  width: 100%;
}

.nameInput {
  margin-right: 12px;
  width: 100%;
}

.emailInput {
  width: 100%;
}

.label {
  color: #42434d;
  font-size: 15px;
  margin-bottom: 3px;
}

.stripeStyledInput {
  border-radius: 5px;
  border: 1.5px solid grey;
  border: 1px solid #e6e6e6;
  box-shadow: 0px 1px 1px rgb(0 0 0 / 3%), 0px 3px 6px rgb(0 0 0 / 2%);
  box-sizing: border-box;
  flex-grow: 1;
  margin-bottom: 15px;
  margin-right: 20px;
  padding: 10px;
  width: 100%;
}

.submitButton {
  align-items: center;
  background-color: #9d9d9d;
  border-radius: 8px;
  color: white;
  cursor: pointer;
  display: flex;
  font-weight: bold;
  height: 40px;
  justify-content: center;
  width: 150px;
}

.submitButton:hover {
  color: #feccd7;
}

@media (max-width: 660px) {
  .paymentForm {
    width: auto;
  }
}
