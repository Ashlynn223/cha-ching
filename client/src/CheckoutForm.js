import {
  PaymentElement,
  useStripe,
  useElements,
} from "@stripe/react-stripe-js";
import "./CheckoutForm.css";
import { useLocation } from "react-router-dom";
const production = "https://zen-blossom.herokuapp.com";
const development = "http://localhost:3000";
const CLIENT_BASE_URL =
  process.env.NODE_ENV === "production" ? production : development;

function CheckoutForm() {
  const { state } = useLocation();
  const subscriptionType = state.subscriptionType;
  const subscriptionPrice = state.subscriptionPrice;
  const stripe = useStripe();
  const elements = useElements();

  console.log(state.subscriptionPrice);

  const capitalize = (str) => str.charAt(0).toUpperCase() + str.slice(1);

  const handleSubmit = async (e) => {
    e.preventDefault();
    await stripe
      .confirmPayment({
        elements,
        confirmParams: {
          return_url: `${CLIENT_BASE_URL}/success`,
        },
      })
      .then((data) => console.log("What am I (data):", data))
      .catch((err) => console.log("I am an error", err));
  };

  return (
    <div className="checkoutFormBackground">
      <div className="header">
        {`${capitalize(subscriptionType)} Subscription`}
      </div>
      <div className="price">{`$${subscriptionPrice} / mo`}</div>
      <div className="paymentFormContainer">
        <form name="payment-form" className="paymentForm">
          <div className="paymentElementcontainer">
            <div className="nameAndEmailInputsContainer">
              <div className="nameAndEmailInputs">
                <div className="nameInput">
                  <div className="label">Name</div>
                  <input className="stripeStyledInput" />
                </div>
                <div className="emailInput">
                  <div className="label">Email</div>
                  <input className="stripeStyledInput" />
                </div>
              </div>
            </div>
            <PaymentElement />
          </div>
          <button onClick={handleSubmit} className="button">
            Submit Payment
          </button>
        </form>
      </div>
    </div>
  );
}

export default CheckoutForm;
