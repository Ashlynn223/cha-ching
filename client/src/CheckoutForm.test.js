import CheckoutForm from "./CheckoutForm";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";

const stripePromise = loadStripe(process.env.REACT_APP_PUBLIC_STRIPE_KEY);

test("renders", async () => {
  const component = render(
    <MemoryRouter
      initialEntries={[
        {
          pathname: "/checkout",
          state: { subscriptionType: "Super Deluxe VIP" },
        },
      ]}
    >
      <Elements stripe={stripePromise}>
        <CheckoutForm />
      </Elements>
    </MemoryRouter>,
  );
  expect(
    await screen.findByRole("button", { name: "Submit Payment" }),
  ).toBeInTheDocument();
  expect(
    await screen.findByText("Super Deluxe VIP Subscription"),
  ).toBeInTheDocument();
  expect(component.asFragment()).toMatchSnapshot();
});
