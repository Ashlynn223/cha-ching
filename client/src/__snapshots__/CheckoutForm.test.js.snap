// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="checkoutFormBackground"
  >
    <div
      class="checkoutFormHeader"
    >
      Super Deluxe VIP Subscription
    </div>
    <div
      class="price"
    >
      $66 / mo
    </div>
    <div
      class="paymentFormContainer"
    >
      <form
        class="paymentForm"
        name="payment-form"
      >
        <div
          class="paymentElementcontainer"
        >
          <div
            class="nameAndEmailInputsContainer"
          >
            <div
              class="nameAndEmailInputs"
            >
              <div
                class="nameInput"
              >
                <div
                  class="label"
                >
                  Name
                </div>
                <input
                  class="stripeStyledInput"
                />
              </div>
              <div
                class="emailInput"
              >
                <div
                  class="label"
                >
                  Email
                </div>
                <input
                  class="stripeStyledInput"
                />
              </div>
            </div>
          </div>
          <div />
        </div>
        <button
          class="submitButton"
        >
          Submit Payment
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
